internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go

```diff
@@ -48,18 +51,39 @@ func newSecureStorage(config *config.AWSConfig, automaticCertificateManagementCo
 		return nil, err
 	}
 	return &secureStorage{
-		secretClient: secretClient,
-		secretCache:  secretCache,
-		secretPrefix: config.SecretManager.SecretPrefix,
+		secretClient:     secretClient,
+		secretCache:      secretCache,
+		storageLock:      sync.Mutex{},
+		storageItemLocks: map[string]*sync.Mutex{},
+		secretPrefix:     config.SecretManager.SecretPrefix,
 	}, nil
 }
 
 func (storage *secureStorage) Lock(ctx context.Context, key string) error {
-	return nil // NOOP as AWS Secret manager uses versioning mechanism to achieve optimistic locking
+	mu, ok := storage.storageItemLocks[key]
+	if !ok {
+		// initialize the storage item lock the first time if it does not exist
+		storage.storageLock.Lock()
+		defer storage.storageLock.Unlock()
+		mu = &sync.Mutex{}
+		storage.storageItemLocks[key] = mu
+		mu.Lock()
```

Comment:
This piece of code confuses me a bit: in a previous comment, we said that this method is  guaranteed to be called synchronously, here however, we are calling 
```go
		storage.storageLock.Lock()
		defer storage.storageLock.Unlock()
```
Unless I miss something that is used to protect against race conditions.
These locks are useless if this is guaranteed to be called synchronously.
If this is not guaranteed to be called synchronously, we can incur the race issue discussed in the previous comment.

Reply:
Thanks @ziccardi I've added a comment https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1639#discussion_r1137851529. 
I'll make sure that this is thread-safe.